src/DDTrace/Integrations/DatabaseIntegrationHelper.php

```diff
@@ -44,34 +44,61 @@ public static function injectDatabaseIntegrationData(HookData $hook, $backend, $
             }
 
             $span = $hook->span();
-            $databaseService = $span->meta['peer.service'] ?? $span->service;
-
-            $query = self::propagateViaSqlComments($hook->args[$argNum], $databaseService, $propagationMode);
+            $targetHost = $span->meta[Tag::TARGET_HOST] ?? '';
+            $dbName = $span->meta[Tag::DB_NAME] ?? $span->meta[Tag::DB_INSTANCE] ?? '';
+            $peerService = $span->meta[Tag::PEER_SERVICE] ?? '';
+
+            $query = self::propagateViaSqlComments(
+                $hook->args[$argNum],
+                $span->service,
+                $propagationMode,
+                $targetHost,
+                $dbName,
+                $peerService
+            );
             $hook->args[$argNum] = $query;
             $hook->overrideArguments($hook->args);
             if ($propagationMode == \DDTrace\DBM_PROPAGATION_FULL) {
-                $hook->span()->meta["_dd.dbm_trace_injected"] = "true";
+                $hook->span()->meta[Tag::DBM_TRACE_INJECTED] = "true";
             }
         }
     }
 
-    public static function propagateViaSqlComments($query, $databaseService, $mode = \DDTrace\DBM_PROPAGATION_FULL)
-    {
+    public static function propagateViaSqlComments(
+        $query,
+        $databaseService,
+        $mode = \DDTrace\DBM_PROPAGATION_FULL,
+        $targetHost = '',
+        $dbName = '',
+        $peerService = ''
+    ) {
         $rootSpan = \DDTrace\root_span();
 
         // Note: the order of the tags is relevant, they must be passed ordered alphabetically
         $tags = [];
 
         if ($databaseService != "") {
-            $tags["dddbs"] = $databaseService;
+            $tags[Tag::DBM_DATABASE_SERVICE_NAME] = $databaseService;
+        }
+
+        if ($targetHost != "") {
+            $tags[Tag::DBM_PEER_HOSTNAME] = $targetHost;
+        }
```

Comment:
This is not visible anymore given that you put these as constants, but this is "ddh", which is not respecting the order (before is "dddbs", next is "dddb"), see the // Note a few lines above.
Hence, to make this much more visible, I would recommend _not_ using Tag:: here, but just adding it as a comment, e.g.
```
// peer hostname
$tags["ddh"] = $targetHost;
```
The strength of constants is when they are reused, but here's no reuse at all, so I think the approach with the comment would be better.

Reply:
> which is not respecting the order

_Wasn't aware there was an order_

Reply:
I'll make the changes

Reply:
It may also make sense to have a test actually verify these are ordered. At least you missed the comment in the source file :-D Others might too.